internal/controller/mariadb_controller.go

```diff
@@ -341,6 +348,55 @@ func (r *MariaDBReconciler) reconcileStatefulSet(ctx context.Context, mariadb *m
 	return ctrl.Result{}, nil
 }
 
+func (r *MariaDBReconciler) reconcilePodLabels(ctx context.Context, mariadb *mariadbv1alpha1.MariaDB) (ctrl.Result, error) {
+	if !mariadb.Replication().Enabled {
+		return ctrl.Result{}, nil
+	}
+
+	podList := corev1.PodList{}
+	listOpts := &client.ListOptions{
+		LabelSelector: klabels.SelectorFromSet(
+			labels.NewLabelsBuilder().
+				WithMariaDBSelectorLabels(mariadb).
+				Build(),
+		),
+		Namespace: mariadb.GetNamespace(),
+	}
+	if err := r.List(ctx, &podList, listOpts); err != nil {
+		return ctrl.Result{}, fmt.Errorf("error listing Pods: %v", err)
+	}
+
+	for _, pod := range podList.Items {
+		var role = "replica"
+
+		if pod.Status.PodIP == "" || pod.Spec.NodeName == "" {
+			continue
+		}
+		podIndex, err := sts.PodIndex(pod.Name)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error getting Pod '%s' index: %v", pod.Name, err)
+		}
+
+		if *podIndex == *mariadb.Status.CurrentPrimaryPodIndex {
+			role = "primary"
+		}
+		pod.Labels = labels.NewLabelsBuilder().WithLabels(pod.Labels).WithPodRole(role).Build()
+
+		if mdbpod.PodReady(&pod) {
+			if err := r.Update(ctx, &pod); err != nil {
```

Comment:
The operator does not have permission for updating `Pods`. Would you mind updating this to automatically generate them when running `make gen`?

https://github.com/mariadb-operator/mariadb-operator/blob/1d6dac00e10ae0fbfbf0fa8cbced5bf0c97266aa/internal/controller/mariadb_controller.go#L98C1-L99C1

Also, since the `Pod` spec is quite big and to follow the codebase conventions, I think it will be better to use a `patch` operation instead, see an example here:
https://github.com/mariadb-operator/mariadb-operator/blob/1d6dac00e10ae0fbfbf0fa8cbced5bf0c97266aa/pkg/controller/galera/recovery.go#L475

We will need to update the helm chart RBAC as well, as it is not automatically generated by `make gen`:
https://github.com/mariadb-operator/mariadb-operator/blob/1d6dac00e10ae0fbfbf0fa8cbced5bf0c97266aa/deploy/charts/mariadb-operator/templates/rbac.yaml#L92
https://github.com/mariadb-operator/mariadb-operator/blob/1d6dac00e10ae0fbfbf0fa8cbced5bf0c97266aa/deploy/charts/mariadb-operator/templates/rbac-namespace.yaml#L75

Reply:
Thanks good catch, I've changed it to use `patch` instead of `update` and updated Helm charts to include patch verb.